workbench/test_utils.py

```diff
@@ -0,0 +1,146 @@
+                    args = list(args)
+                    args.insert(1, xblock)
+
+            return func(*args, **kwargs)
+        return _wrapped
+    return _decorator
+
+
+class XBlockHandlerTestCaseMixin(object):
+    """Load the XBlock in the workbench runtime to test its handler. """
+
+    def setUp(self):
+        """Create the runtime. """
+        super(XBlockHandlerTestCaseMixin, self).setUp()
+        self.runtime = WorkbenchRuntime()
+
+    def set_user(self, user_id):
+        """Provide a user ID to the runtime.
+
+        Args:
+            user_id (str): a user ID.
+
+        Returns:
+            None
+        """
+        self.runtime.user_id = user_id
+
+    def load_scenario(self, xml_path):
+        """Load an XML definition of an XBlock and return the XBlock instance.
+
+        Args:
+            xml (string): Path to an XML definition of the XBlock, relative
+                to the test module.
+
+        Returns:
+            XBlock
+        """
+        block_id = self.runtime.parse_xml_string(
+            self.load_fixture_str(xml_path), self.runtime.id_generator
+        )
+        return self.runtime.get_block(block_id)
+
+    def request(self, xblock, handler_name, content, request_method="POST", response_format=None):
+        """Make a request to an XBlock handler.
+
+        Args:
+            xblock (XBlock): The XBlock instance that should handle the request.
+            handler_name (str): The name of the handler.
+            content (unicode): Content of the request.
+
+        Keyword Arguments:
+            request_method (str): The HTTP method of the request (defaults to POST)
+            response_format (None or str): Expected format of the response string.
+                If `None`, return the raw response content; if 'json', parse the
+                response as JSON and return the result.
+
+        Raises:
+            NotImplementedError: Response format not supported.
+
+        Returns:
+            Content of the response (mixed).
+        """
+        # Create a fake request
+        request = webob.Request(dict())
+        request.method = request_method
+        request.body = content
+
+        # Send the request to the XBlock handler
+        response = self.runtime.handle(xblock, handler_name, request)
+
+        # Parse the response (if a format is specified)
+        if response_format is None:
+            return response.body
+        elif response_format == 'json':
+            return json.loads(response.body)
+        else:
+            raise NotImplementedError("Response format '{format}' not supported".format(response_format))
+
+    @staticmethod
+    def load_fixture_str(path):
+        """Load data from a fixture file.
+
+        Args:
+            path (str): Path to the file.
+
+        Returns:
+            unicode: contents of the file.
+        """
+        base_dir = os.path.dirname(os.path.abspath(__file__))
```

Comment:
@wedaly we need to fix this. Currently we are getting `base_dir` of `xblock-sdk` (due to `__file__` path) not the `ed-reverification-xblock` on which `scenario` decorator is used.
We have to modify this to get base path of file of that `xblock`'s  test directory on which `scenario` decorator is used.
Should we remove `base_dir` path finding and pass absolute url in `scenario` decorator?